helper/resource/testing.go

```diff
@@ -559,6 +559,12 @@ func Test(t TestT, c TestCase) {
 		providers[name] = p
 	}
 
+	if acctest.TestHelper != nil {
+		// inject providers for ImportStateVerify
+		RunLegacyTest(t.(*testing.T), c, providers)
```

Comment:
the providers map is something I managed to cut as cruft in the version2 branch, wondering if we can get away without it? Can't remember if I authored this comment but seems like its lingering around for ImportStateVerify

Reply:
I suppose we can keep this in when merging into `master` (as opt-in) and then remove it in `version2` branch. This would align with the risk profile you set by removing it in `version2` I expect?

Reply:
I agree we should remove the conditional check that acts as an "opt-in" in `version2`. Unfortunately it's the map passed to `RunLegacyTest` called `providers` that I cut from `version2` and once this is merged to master I will need to rebase `version2` and restore it. Unless of course we trace `providers` through `RunLegacyTest` and find a way to remove it.

It's not a big deal I'm sure we will get the merge conflicts solved one way or another.